### PR TITLE
127 setup devise confirmation and notification mails

### LINKED ISSUE #127: Setup devise confirmation and notification mails
Send confirmation and notification mails for all devise modules we want to use.

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #127.

### CODE REVIEW

File: spec/controllers/v2/users/passwords_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      end
+
+      it 'sends an instructions email and a notification email' do
+        expect(UsersMailer.deliveries.size).to eq(2)
+      end
+
+      context 'confirmation mail' do
+        let(:email) { emails[0] }
+
+        it 'is has the correct recepient' do
```
Comment (ebolloff) [inline spec/controllers/v2/users/passwords_controller_spec.rb]: Recipient

File: spec/controllers/v2/users/passwords_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          link = %r{<a href="http://example.test/users/password\?reset_password_token=[^"]+">Change my password</a>}
+          # rubocop:enable Metrics/LineLength
+          expect(email.body.encoded).to match(link)
+        end
+      end
+
+      context 'password changed notification email' do
+        let(:email) { emails[1] }
+
+        it 'has the correct recepient' do
```
Comment (ebolloff) [inline spec/controllers/v2/users/passwords_controller_spec.rb]: Copy paste error

File: spec/controllers/v2/users/registrations_controller_spec.rb [hunk 3]
```diff
@@ -16,16 +16,50 @@
...
       end
       it { expect(response).to have_http_status(:created) }
       it { |example| expect([example, response]).to comply_with_api }
+
+      context 'confirmation mail' do
+        it 'sends an email' do
+          expect(UsersMailer.deliveries.size).to eq(1)
+        end
+
+        it 'is has the correct recepient' do
```
Comment (ebolloff) [inline spec/controllers/v2/users/registrations_controller_spec.rb]: Again copy paste error

File: spec/controllers/v2/users/registrations_controller_spec.rb [hunk 4]
```diff
@@ -57,20 +94,105 @@
...
+        end
+
+        it 'sends a confirmation email and two notification emails' do
+          expect(UsersMailer.deliveries.size).to eq(3)
+        end
+
+        context 'email changed notification email' do
+          let(:email) { emails[0] }
+
+          it 'has the correct recepient' do
```
Comment (ebolloff) [inline spec/controllers/v2/users/registrations_controller_spec.rb]: ....

File: spec/support/api/schemas/apidoc/unlock.json [hunk 5]
```diff
@@ -0,0 +1,82 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Unlocking",
+  "description": "Endpoint to unlock a locked user account",
+
+  "links": [
+    {
+      "title": "Unlock an account",
```
Comment (ebolloff) [inline spec/support/api/schemas/apidoc/unlock.json]: This has to be a `PATCH`, confirmation as well.

**Comment by eugenk** [comment]
I rebased on master and put the new onothub-models gem into the Gemfile[.lock] by commit-amend on the first commit. Please approve again.

**Pull request merged**